src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -81,42 +81,46 @@ public ITest Build(Assembly assembly, IDictionary<string, object> options)
 #endif
 
             string assemblyPath = AssemblyHelper.GetAssemblyPath(assembly);
-            return Build(assembly, assemblyPath, options);
+            string assemblyName = assemblyPath.IndexOfAny(Path.GetInvalidPathChars()) == -1
+                ? Path.GetFileName(assemblyPath)
+                : AssemblyHelper.GetAssemblyName(assembly).FullName;
+
+            return Build(assembly, assemblyName, options);
         }
 
         /// <summary>
         /// Build a suite of tests given the filename of an assembly
         /// </summary>
-        /// <param name="assemblyName">The filename of the assembly from which tests are to be built</param>
+        /// <param name="assemblyNameOrPath">The filename or path of the assembly from which tests are to be built</param>
         /// <param name="options">A dictionary of options to use in building the suite</param>
         /// <returns>
         /// A TestSuite containing the tests found in the assembly
         /// </returns>
-        public ITest Build(string assemblyName, IDictionary<string, object> options)
+        public ITest Build(string assemblyNameOrPath, IDictionary<string, object> options)
         {
 #if NETSTANDARD1_4
-            log.Debug("Loading {0}", assemblyName);
+            log.Debug("Loading {0}", assemblyNameOrPath);
 #else
-            log.Debug("Loading {0} in AppDomain {1}", assemblyName, AppDomain.CurrentDomain.FriendlyName);
+            log.Debug("Loading {0} in AppDomain {1}", assemblyNameOrPath, AppDomain.CurrentDomain.FriendlyName);
 #endif
 
             TestSuite testAssembly = null;
 
             try
             {
-                var assembly = AssemblyHelper.Load(assemblyName);
-                testAssembly = Build(assembly, assemblyName, options);
+                var assembly = AssemblyHelper.Load(assemblyNameOrPath);
+                testAssembly = Build(assembly, Path.GetFileName(assemblyNameOrPath), options);
             }
             catch (Exception ex)
             {
-                testAssembly = new TestAssembly(assemblyName);
+                testAssembly = new TestAssembly(Path.GetFileName(assemblyNameOrPath));
                 testAssembly.MakeInvalid(ExceptionHelper.BuildMessage(ex, true));
             }
 
             return testAssembly;
         }
 
-        private TestSuite Build(Assembly assembly, string assemblyPath, IDictionary<string, object> options)
+        private TestSuite Build(Assembly assembly, string assemblyName, IDictionary<string, object> options)
```

Comment:
I'm thinking this would be most clear as `string suiteName`, and `suiteName` should also be the name of the `BuildTestAssembly` parameter.